addons/config/external-dns.yaml

```diff
@@ -0,0 +1,32 @@
+## This controls which types of resource external-dns should 'watch' for new
+## DNS entries.
+sources:
+  - service
+  - ingress
+## Limit possible target zones by domain suffixes (optional)
+#         - --domain-filter=mgmt.{{ .Values.domain }}
+#        - --domain-filter=app.{{ .Values.domain }}
+# TODO Both the fixes in this file require an understanding of the mgmt and app domains, host names and certificates being created
+# TODO Fix target domain suffixes
+domainFilters: ["mgmt.easi-eo.solutions","app.easi-eo.solutions","easi-eo.solutions"]
```

Comment:
Domain filters will be different for every user, therefore they should be set in terraform using user defined variables.

Reply:
Yep, that is why there is a TODO on them ;-)

I'm not sure how to approach getting terraform to either a) update the yaml or b) just set these values in the .tf file, and leave the static values in the yaml - I believe the two will be combined together but haven't tried it yet.

Also, the example I have in there suggest we have mgmt and app domains that are managed by the datacube-k8s-eks setup including certs for the standard included fleet of ODC tools, PLUS a user defined domain which can be managed separately.
In my current deployment that means I have app, mgmt and hub prefixes each with their own cert and all managed by ingress rules and external-dns. My *.easi-eo.solutions cert is for the entire *.easi-eo.solutions domain. I thought at first the two additional certs might cause a conflict but apparently they don't. I'm not sure if you know how this works better than I but I'm assuming if a more precise cert matches it uses that rather than using the general one but I've not placed any services under the other two prefixes yet so please advise.

If this all makes sense then it should be a simple matter of a few terraform vars and off we go.

Reply:
The domain should just be set in the .tfvars and passed through variables to this helm chart. The only person that can run the current code is the owner of easy-eo.solutions, so it doesn't work as a default variable in the config file, (happy for it to be default in the the .tfvars with a comment it must be changed though).